src/toil/jobStores/aws/jobStore.py

```diff
@@ -1142,11 +1142,11 @@ def download(self, localFilePath):
             elif self.version:
                 headers = self._s3EncryptionHeaders()
                 key = self.outer.filesBucket.get_key(bytes(self.fileID), validate=False)
-                for attempt in retry_s3():
-                    with attempt:
-                        key.get_contents_to_filename(localFilePath,
-                                                     version_id=self.version,
-                                                     headers=headers)
+                with open(localFilePath, 'wb') as fileObject:
+                    handler = boto.s3.resumable_download_handler.ResumableDownloadHandler(num_retries=10)
+                    handler.get_file(key, fileObject, headers, version_id=info.version)
+                if handler.content_length != self.fileID.size:
```

Comment:
I don't see a `content_length` property on the ResumableDownloadHandler documented at https://www.pydoc.io/pypi/boto-2.42.0/autoapi/s3/resumable_download_handler/index.html#s3.resumable_download_handler.ResumableDownloadHandler or implemented at https://chromium.googlesource.com/external/boto/+/77d18c583b43265962e628d8597713841e7dfd62/boto/s3/resumable_download_handler.py so I don't think this check is going to work. Since the resumable download handler handles all the resuming, I think it will just raise its own error if the whole file can't be downloaded, so you might not need this check.